src/application/extension-popup-menu/app.tsx

```diff
@@ -0,0 +1,55 @@
+import { Route, Routes } from 'react-router-dom';
+
+import { useExtensionSettings } from 'shared/extension';
+import { Closable } from 'shared/ui';
+
+import { Footer } from './footer';
+import { PopupHomeView } from './popup-home-view';
+import { TopBar } from './top-bar';
+import { CustomizationSettings } from './settings-view/customization-settings';
+import { MainSettingsMenu } from './settings-view/main-settings-menu';
+
+export const EXTENSION_POPUP_MENU_ROUTES = {
+  HOME: '/',
+  SETTINGS: {
+    MAIN_VIEW: '/settings',
+    CUSTOMIZATION: '/settings/customization',
+  },
+};
+
+export const App = () => {
+  const { isPopupMenuVisible, hidePopupMenu } = useExtensionSettings();
+
+  if (!isPopupMenuVisible) {
+    return null;
+  }
+  return (
+    <Closable
+      closeButtonClassName="hidden"
+      closeOnClickAway
+      onClose={hidePopupMenu}
+      className="fixed right-6 top-6 z-[9999] flex size-[570px] flex-col overflow-hidden rounded-md bg-white p-0 shadow-lg"
+    >
+      <TopBar />
+
+      <div className="max-h-[460px] grow overflow-y-auto bg-gray-100 [scrollbar-color:gray_#b7b7b7] [scrollbar-width:thin]">
+        <Routes>
+          <Route
+            path={EXTENSION_POPUP_MENU_ROUTES.HOME}
+            element={<PopupHomeView />}
+          />
+          <Route
+            path={EXTENSION_POPUP_MENU_ROUTES.SETTINGS.MAIN_VIEW}
+            element={<MainSettingsMenu />}
+          />
+          <Route
+            path={EXTENSION_POPUP_MENU_ROUTES.SETTINGS.CUSTOMIZATION}
+            element={<CustomizationSettings />}
+          />
+        </Routes>
```

Comment:
there is new recommended way of defining routes in react:
https://reactrouter.com/en/main/routers/create-browser-router

you could move this router definition outside to `router.tsx`:
```tsx
export const extensionPopupRouter = createBrowserRouter(...);
```

and inside here only do:
```ts
<RouterProvider router={extensionPopupRouter} />
```

but that's minor thing actually

Reply:
one thing that I am worried about is that we won't be able to redirect popup to another view from another module like polymarket because you will be outside of react router context

Ideally I would want to do this inside polymarket:
```tsx
const logIn = () => {
  popup.open(EXTENSION_POPUP_ROUTE.LOGIN)
}

<Button onClick={login}>Connect wallet</Button>
```

Let's maybe discuss it internally in discord and leave for later

Reply:
As discussed on Discord, I've replaced the HashRouter with the MemoryRouter and moved it higher in our Application hierarchy